charts/hivemq-platform-operator/templates/rbac.yml

```diff
@@ -104,8 +104,8 @@ rules:
   - apiGroups:
       - rbac.authorization.k8s.io
     resources:
+      - roles
       - rolebindings
-      - clusterroles
```

Comment:
I think we only need these custom permissions if we want to create/update/delete resources. We should be able to get/list all resources with the `view` ClusterRole that we are binding here: https://github.com/hivemq/helm-charts/blob/master/charts/hivemq-platform-operator/templates/bindings.yml#L10

Reply:
You might be right, I just looked up that default role and it doesn't include roles and bindings:

> view: Allows read-only access to see most objects in a namespace. It does not allow viewing roles or role bindings.
This role does not allow viewing Secrets, since reading the contents of Secrets enables access to ServiceAccount credentials in the namespace, which would allow API access as any ServiceAccount in the namespace (a form of privilege escalation).

https://kubernetes.io/docs/reference/access-authn-authz/rbac/#user-facing-roles

Reply:
I guess we need some integration tests in here to check these permissions.
We might have a custom ServiceAccount with Role and RoleBinding or ClusterRole and ClusterRoleBinding. I'm not even sure why the existing `HelmCustomServiceAccountIT` doesn't go into an error state (that custom service account shouldn't have any permissions and there is a check for that).

Reply:
I've updated the section to configure full CRUD permissions for roles and rolebindings, and get, list, watch for clusterroles and clusterrolebindings. That should be the minimum we need.

Reply:
BTW, didn't check out myself, but I'm curious then why the `HelmCustomServiceAccountIT` didn't fail when you removed the `clusterroles` permission?

Reply:
In general `HelmCustomServiceAccountIT` works because the Platform Operator creates the permissions (`Role` and `RoleBinding`) for the configured `ServiceAccount`, no matter if it's a custom or the default `ServiceAccount`.

I've also added `clusterroles` and `clusterrolebindings` as read permissions.

Reply:
I just gave it a spin with `platformServiceAccount.permissions.create=false` and no RBAC for:
```yaml
  - apiGroups:
      - rbac.authorization.k8s.io
    resources:
      - clusterroles
      - clusterrolebindings
    verbs:
      - get
      - list
      - watch
```
This results in an `ERROR/INVALID_CUSTOM_RESOURCE_SPEC` state with the message:
```
The custom resource spec is invalid: Failure executing: GET at: https://10.96.0.1:443/apis/rbac.authorization.k8s.io/v1/clusterrolebindings. Message: clusterrolebindings.rbac.authorization.k8s.io is forbidden: User "system:serviceaccount:default:hivemq-platform-operator-test-operator" cannot list resource "clusterrolebindings" in API group "rbac.authorization.k8s.io" at the cluster scope. Received status: Status(apiVersion=v1, code=403, details=StatusDetails(causes=[], group=rbac.authorization.k8s.io, kind=clusterrolebindings, name=null, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=clusterrolebindings.rbac.authorization.k8s.io is forbidden: User "system:serviceaccount:default:hivemq-platform-operator-test-operator" cannot list resource "clusterrolebindings" in API group "rbac.authorization.k8s.io" at the cluster scope, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=Forbidden, status=Failure, additionalProperties={}).
```
I think that's fine, since it will only occur if you mess with the Helm chart in an unsupported way.

Reply:
I gave it another spin with the actual branches and `platformServiceAccount.create=false` and `platformServiceAccount.permissions.create=false`, but activated validation. Using the example K8s resource scripts, I could go step by step through the different error stages of "service account is missing" -> adding service account -> "no permissions" -> adding rolebinding -> "rolebinding roleRef role is missing" -> adding role -> RUNNING.